### DISCUSSION TITLE
How do I do a new install of 8.x to SQL Server

### DISCUSSION THREAD

**Discussion by natechadwick** [percussion/percussioncms#709] [Q&A]
The new install of 8.x seems to only support the embedded Derby database.  How do I get it to work with SQL Server instead?

**Comment by natechadwick** [comment]
The new install still isn't handling other database types, but I think this approach should work.

Edit the {InstallDir}/jetty/base/etc/perc-ds.properties file with the SQL Server connection info.  We use the Microsoft JDBC driver now, not the jtds driver as it wasn't being maintained with new SQL Server versions. 

Example {InstallDir}/jetty/base/etc/perc-ds.properties:

```
perc.conn.1.db=yourdb
perc.conn.1.default=Y
perc.conn.1.ds=jdbc/RhythmyxData
perc.conn.1.name=RhythmyxData
perc.conn.1.schema=dbo
perc.ds.1.connections.max=1000
perc.ds.1.connections.min=30
perc.ds.1.driver.class=com.microsoft.sqlserver.jdbc.SQLServerDriver
perc.ds.1.driver.name=sqlserver
perc.ds.1.idle.ms=59000
perc.ds.1.max.lifetime=1800000
perc.ds.1.name=jdbc/RhythmyxData
perc.ds.1.pwd=yourpwd
perc.ds.1.pwd.encrypted=N
# replace with your instance info for dev using trustServerCertificate=true resolves tls issues 
# with self signed sql server cert
perc.ds.1.server=//127.0.0.1\:1433;trustServerCertificate=true
perc.ds.1.uid=your uid/sa

```

As long as you have a java 1.8 JRE installed on your machine, you should be able to re-run the installer with the distribution jar (assuming windows?).

- Start command prompt as administrator.
- java -jar perc-distribution-tree.jar c:\pathtoyoruinstall

This should run in upgrade mode and install the schema on SQL Server, switching from Derby.

**Reply by davidmrowe** [reply]
Hi.  Where would I find the perc-distribution-tree.jar file? I installed 8.0.3 from perc-installer-build-8.0.3-windows64.zip and the setup.exe from within that?

Thanks in advance.

**Reply by natechadwick** [reply]
Hi @davidmrowe,

The distribution tree jar is on the download page on the support portal, it is also on the release page:  https://github.com/percussion/percussioncms/releases/tag/v8.1.2

The graphical installer is a minimal front end that invokes the java -jar perc-distribution-tree.jar <installDir> command.  You do need a Java 1.8 runtime installed.  Installing with later java vesion's may cause the installation to fail.
